lib/internal/streams/pipeline.js

```diff
@@ -197,12 +198,12 @@ function pipeline(...streams) {
         const pt = new PassThrough();
         if (isPromise(ret)) {
           ret
+            .catch((err) => {
+              pt.destroy(err);
+            })
             .then((val) => {
+              value = val;
               pt.end(val);
```

Comment:
This means that `.end` will be called and `value` will be undefined. This is a bit confusing.

Reply:
In what way?

Reply:
@benjamingr: Since you approved the PR I assume this is not blocking for you?

Reply:
@ronag I think @benjamingr meant the specifics of Promise handling here => the current solution in case of error will call `pt.destroy(err)` and then `pt.end(undefined)` due to the fact that `.catch` handles error and returns undefined to be processed by `.then`.
Perhaps you meant to use
```javascript
ret.then((val) => {
  value = val;
  pt.end(val);
}, (err) => {
  pt.destroy(err);
})
```
to call  **only** `destroy` in case of error?

Reply:
@lundibundi: Thank's a lot! I understand the confusion. Resolved according to your suggestion.

Reply:
Yes, exactly. It wasn't wrong it was just confusing :]